### PR TITLE
Fixing count caught pokemons per region

### PR DESCRIPTION
**Pull request by Jaaslet**
This is an attempt to fix how number of caught pokemons per region is caught.

- [x] Fix it for pokedex-achievements
- [x] Fix it for all caught challenge
- [x] Look for other places it is needed

### CODE REVIEW

**Comment by Aegyo** [comment]
Could you describe the issue this is fixing in more detail please?

**Comment by Ultima1990** [comment]
> Could you describe the issue this is fixing in more detail please?

The issue is Alola related. Regional forms are currently being counted for the Alola dex completion achievement, but the number of mons required is still based on base mons. This means the achievement is awarded prematurely.

File: src/scripts/pokemons/PokemonHelper.ts [hunk 1]
```diff
@@ -387,4 +387,34 @@ class PokemonHelper {
...
     }
+
+    public static numberOfPokemonsCaughtWithBaseFormNativeToRegion(region: GameConstants.Region) {
+        let result = 0;
+        let currentPokemonThisRegion = false;
+        let currentPokemonCaught = false;
+        let lastPokemonBaseId = 0;
+
+        // Counts each form once, and only if the baseform is native to this region
+        App.game.party.caughtPokemon.sort((a, b) => a.id - b.id).forEach(pokemon => {
```
[the remark this answers is not in the document]
Comment (Jaaslet) [reply src/scripts/pokemons/PokemonHelper.ts:398]: Good catch! Should be fixed now

**Comment by RedSparr0w** [comment]
I think this PR is trying to correct the issue the opposite way around?

We are wanting the achievement to check that the player has every possible mon (excluding variants), not just base forms.

With this PRs code it shows I only have acquired `34` mons, with the old code it shows `43 `
As far as I can tell `43` is the value we want from this though?

<details>
<summary>Actual mons:</summary>

image
</details>

I think the problem lies in `PokemonHelper.calcUniquePokemonsByRegion(region)`

**Comment by Jaaslet** [comment]
From a poll, in the pokecrew-chat, it shows most want an other solution for the achievement being awarded before time.
This solution was to filter out all new variants, so only new pokemons was required, for the achievement/catch all challenge.
The poll shows mosts people want regional variants to be included in the achievement/challenge requirements.
I will close this pull request

**Pull request closed without merging**